extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.cloud.google;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.inject.AbstractModule;
+import com.google.inject.BindingAnnotation;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import java.io.IOException;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import org.dataportabilityproject.api.launcher.ExtensionContext;
+
+/** Bindings for cloud platform components using Google Cloud Platform. **/
+final class GoogleCloudModule extends AbstractModule {
+  // The value for the 'cloud' flag when hosting on Google Cloud Platform.
+  private static final String GOOGLE_CLOUD_NAME = "GOOGLE";
+  // Environment variable where GCP project ID is stored. The value is set in
+  // config/k8s/api-deployment.yaml.
+  private static final String GCP_PROJECT_ID_ENV_VAR = "GOOGLE_PROJECT_ID";
+  // Environment variable for path where GCP credentials are stored. The value of the environment
+  // variable (i.e. the path to creds) is configured in config/k8s/api-deployment.yaml. The creds
+  // themselves are exposed as a Kubernetes secret.
+  private static final String GCP_CREDENTIALS_PATH_ENV_VAR = "GOOGLE_APPLICATION_CREDENTIALS";
+  // The path where Kubernetes stores Secrets.
+  private static final String KUBERNETES_SECRETS_PATH_ROOT = "/var/secrets/";
+
+  @VisibleForTesting enum Environment {
```

Comment:
Since we already have a concept of environment, do we want to call this Realm? Or SubEnvironment? Or something else?

Reply:
I only saw the definition in Core which I figured we would deprecate. Is there another one yet? I have not rebased your changes yet as to not mess up the code review, but I don't see another Environment in github code search. I will put a TODO for now.